lib/chunks.sql

```diff
@@ -190,27 +190,83 @@ WITH
         AND a.course_instance_id = ci.id
       )
   ),
-  chunks_to_delete AS (
+  -- We use separate CTEs for each chunk type so that Postgres doesn't tye to
+  -- compute a huge cross product of the chunks + metadata on `type` as an
+  -- intermediate step. This was O(n^2) with the number of chunks to delete.
+  question_chunks_to_delete AS (
     SELECT
       id
     FROM
       chunks
       JOIN chunks_metadata AS cm ON (
         chunks.course_id = $course_id
         AND chunks.type = cm.type
-        AND CASE
-          WHEN (
-            cm.type = 'elements'
-            OR cm.type = 'elementExtensions'
-            OR cm.type = 'clientFilesCourse'
-            OR cm.type = 'serverFilesCourse'
-          ) THEN TRUE
-          WHEN (cm.type = 'clientFilesCourseInstance') THEN cm.course_instance_id = chunks.course_instance_id
-          WHEN (cm.type = 'clientFilesAssessment') THEN cm.assessment_id = chunks.assessment_id
-          WHEN (cm.type = 'question') THEN cm.question_id = chunks.question_id
-          ELSE FALSE
-        END
+        AND cm.question_id = chunks.question_id
       )
+    WHERE
+      chunks.type = 'question'
+  ),
+  client_files_course_instance_chunks_to_delete AS (
+    SELECT
+      id
+    FROM
+      chunks
+      JOIN chunks_metadata AS cm ON (
+        chunks.course_id = $course_id
```

Comment:
Is there a reason why this is on the join instead of the where?

Reply:
Nope - I'll move it!

Reply:
Addressed in a442019.